docs/tutorials/podman_pasta_ipv6_with_snid_ipv4.md

```diff
@@ -0,0 +1,761 @@
+
+I am NOT used to this Syntax, so I am just using the Caddyfile, at least for now.
+This is also due to the Fact that I am self-managing the Letsencrypt Certificates using `certbot` and distributing them across my Infrastructure.
+
+Your mileage may vary :).
+
+```
+# Example and Guide
+# https://caddyserver.com/docs/caddyfile/options
+
+# General Options
+{
+    # Debug Mode
+    debug
+
+    # Ports Configuration
+    #http_port 80
+    #https_port 443
+
+    # TLS Options
+    auto_https disable_certs
+
+    # Default SNI
+    default_sni MYDOMAIN.TLD
+}
+
+
+localhost {
+	reverse_proxy /api/* localhost:9001
+}
+
+application01.MYDOMAIN.TLD {
+        tls /certificates/MYDOMAIN.TLD/fullchain.pem /certificates/MYDOMAIN.TLD/privkey.pem
+        log {
+		output file /var/log/application01.MYDOMAIN.TLD/access.json {
+			roll_size 100MiB
+			roll_keep 5000
+			roll_keep_for 720h
+            roll_uncompressed
+		}
+        format json
+	}
+
+    reverse_proxy http://[::1]:8080
+}
+```
+
+# Redirects
+## General
+When handling Redirects, HTTP Codes `301` (`Moved Permanently`), `302` (`Found (Temporary Redirect)`), `307` (`Temporary Redirect`) and `308` (`Permanent Redirect`) are Typically Used.
+
+References:
+- https://storychief.io/blog/301-302-307-308-redirect
+- https://www.infidigit.com/blog/308-permanent-redirect/
+
+When a 308 redirect code is specified, the client must repeat the exact same request (POST or GET) on the target location. For 301 redirect, the client may not necessarily follow the exact same request. 
+
+Since the HTTPS must always be Enforced (`Permanent Redirect`) and the Client should perform the exact same Request over HTTPS that was performed over HTTP, the Code `308` seems more Appropriate.
+
+## HTTP -> HTTPS for IPv4 Redirects
+The `snid` Service does NOT handle HTTP (non-HTTPS) Requests and does NOT bind to the IPv4 Address on Port 80.
+
+This Problem (IPv4 HTTP -> HTTPS Redirects) can easily be solved by using a one-off (for the entire Podman Host) Caddy Container.
+
+(Option 1) `compose.yml` with "Normal" Ports Mappings:
+```
+services:
+  redirect-http-ipv4-caddy:
+    image: caddy:latest
+    pull_policy: "missing"
+    container_name: redirect-http-ipv4-caddy
+    restart: "unless-stopped"
+    security_opt:
+      - no-new-privileges:true
+      - label=type:container_runtime_t
+    ports:
+      - target: 80
+        host_ip: 172.16.1.10
+        published: 80
+        protocol: tcp
+    network_mode: "pasta:--ipv4-only"
+    volumes:
+      - ./Caddyfile:/etc/caddy/Caddyfile:ro,z
+      - ~/containers/data/redirect-http-ipv4-caddy:/data:rw,z
+      - ~/containers/log/redirect-http-ipv4-caddy:/var/log:rw,z
+      - ~/containers/config/redirect-http-ipv4-caddy:/config:rw,z
+      - ~/containers/certificates/letsencrypt:/certificates:ro,z
+    environment:
+      - CADDY_DOCKER_CADDYFILE_PATH=/etc/caddy/Caddyfile
```

Comment:
should also be cleaned.
- volumes, restart, pull_policy, container_name, environment

see compose above

Reply:
Well ... In this case the Certificates is for sure not needed.

The Rest ... well ... Depends how much you want to configure ? 

I think we can agree we have (very) different Approaches (I always prefer an explicit Configuration and keeping `compose.yml` / configuration Files in the same Structure for different Applications/Uses, whereas you prefer to cut everything down custom-made between Applications/Uses and let the Application do its Defaults).

Not a question of right/wrong, just a different Approach. I might be overriding a default with a worse value, whereas you might be a non-sane default to begin with.

I've been bitten by both approaches in the past, that's why I say there is no right/wrong answer really ...